### ISSUE TITLE
Cant seem to make this work

### ISSUE THREAD

**Issue by alanef** [lucmos/UltrawideWindows#28]
Hi, this looks ideal, but I have tried several times and nothing happens when I use key mappings.

Is this related to the issue with 'latest master' perhaps - should I try an earlier version.

Of is there just some easy noob mistake I have missed?

**Comment by Stijn-vdD** [comment]
Hi, I just tried to install the script in a clean Arch VM.
I could replicate the issue when following the instructions in the readme, then the KWin shortcuts are not initialised. 

I did find a method which does make the shortcuts work, which is to use the KDE system settings GUI. Go to Window Management > KWin Scripts and press install from file. Using this method the shortcuts are created and everything seems to work.

@lucmos I think this issue would be resolved by making the script on the kde store up to date and/or changing the instructions in the readme.

Please let me know whether this works for you.

**Comment by alanef** [comment]
I did  try manual install and KDE install - the shortcut were loaded in the KDE GUI but didn't action.

**Comment by Stijn-vdD** [comment]
So the shortcuts are there, the script is shown and even after a restart it does not do anything?

**Comment by Stijn-vdD** [comment]
Alright now that I've restarted my VM it has stopped working, I'll try to find out what is going on.

**Comment by alanef** [comment]
I have just tried it on my lap top - install from KDE store  -  and apply - Shortcut not there UNTIL  I logout  / in - shortcuts then there.

As my laptop doesn't have a Num pad I remapped some keys - and they work.

I suspect that it is an issue with the way my keyboard num pad maps !  Maybe.  I'll test it on the desktop tommorrow by remapping a few keys.   

So it definitely works on my laptop - but not that useful as the screen is on;y 15.1 :)

**Comment by Stijn-vdD** [comment]
Yup sometimes you have to restart the wm to put the changes through. Do note that what's on the store is a very old version so it may behave differently, and now that the shortcuts are there they will stay. 
It would be interesting if it is indeed a problem with the keyboard.

**Comment by lucmos** [comment]
Sorry for the inconvenience and thanks a lot for the debugging. 

Unfortunately, I do not have access to a keyboard with a physical or virtual numpad in this period. I didn't find a way to quickly emulate it and also changing the shortcuts somehow it's not working. 

I was waiting to test it before updating the store version but unfortunately it is taking more than expected.

---

To update the shortcuts is it still working the `scripts/update.sh` script?

**Comment by alanef** [comment]
Hi,

So I tried different versions.

clone of master  did not operate at all

clone  branch 3.0 worked for me

so I tried 4.1  and that also worked.  So currently operational on 4.1 for me.

 ( for anyone reading  and unsure this is how to get a branch )


`git clone --depth 1 --branch 4.1  git@github.com:lucmos/UltrawideWindows.git`

**Comment by lucmos** [comment]
Thanks for the testing! 

Do you have any ideas @Stijn-vdD ?

I will reset master on 4.1, push a new branch with the new changes so that we can debug there if that's ok for you

**Comment by alanef** [comment]
I wish I knew more about how to debug these scripts - but I'm a PHP developer so it is all a bit strange for me. But thank you both, I got an Ultrawide yesterday to replace the old 3 screen set up and your solution is just great.

**Comment by Stijn-vdD** [comment]
Yeah, good idea to set up a debugging branch. I'll see if I can figure something out.

**Comment by lucmos** [comment]
No worries @alanef, I am extremely inexperienced too in Javascript/Kwin scripting

---

@Stijn-vdD there is now a branch `debug/stijn` that is the old master. A `master` and a new `develop` branch are now on the `4.1`.
If you have any time feel free to create a PR from `debug/stjin` into `develop`; once that it is ready we can test develop and merge it into master when it works

**Comment by thackel** [comment]
Thanks that there is some effort to get ultrawides usable with KDE. The most convenient solution would be to integrate this functionality into kwin. I would like to have the possibility to maximize a window in a 3:1 grid by dragging the window to the upper edge similar to the existing "maximize" functionaly.
With this KWin script I have to take my hand of the mouse.

The version 4.0 which is part of the "kde-store" does not work at all.
I had to clone this repository and run the `update.sh` script to get it going.

The install instructions should be updated to:
```
git clone https://github.com/lucmos/UltrawideWindows.git
cd scripts
./update.sh
```

Thanks for your work!

**Comment by lucmos** [comment]
@Stijn-vdD I just tested the branch `debug/stijn` and everything seems to work as expected, with a fresh installation of Manjaro.


@alanef can you still reproduce the problem? I did the following:

- Remove UltrawideWindows from the  KwinScripts settings: "Window Management -> Kwin Scripts"
- `kwin_x11 --replace &` and ensure it does not work
- Then, install the new version:
```bash
git clone git@github.com:lucmos/UltrawideWindows.git
cd UltrawideWindows
git checkout debug/stijn
./scripts/update.sh
```
- `kwin_x11 --replace &` and ensure it works


Please let me know if anyone can reproduce the problem, otherwise, I will update the master with the latest changes in that branch and publish a new release on the kde store.

**Comment by lucmos** [comment]
I released the version `4.2`, feel free to reopen in there are any problems
